system/HotReloader/IteratorFilter.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\HotReloader;
+
+use RecursiveFilterIterator;
+use RecursiveIterator;
+
+/**
+ * @internal
+ *
+ * @extends RecursiveFilterIterator<RecursiveIterator>
```

Comment:
@paulbalandan @kenjis What's the proper lines to have in the docblock here to make our static analysis happy?

Reply:
Something like this? But I don't know the exact types.
```
 * @template-extends RecursiveFilterIterator<TKey, TValue, TIterator>
 * @template-implements RecursiveIterator<TKey, TValue>
 ```

There is some question as to how useful this type information is, so we can suppress errors.

```diff
--- a/system/HotReloader/IteratorFilter.php
+++ b/system/HotReloader/IteratorFilter.php
@@ -17,8 +17,7 @@ use RecursiveIterator;
 /**
  * @internal
  *
- * @extends RecursiveFilterIterator<RecursiveIterator>
- * @implements RecursiveIterator<RecursiveIterator>
+ * @psalm-suppress MissingTemplateParam
  */
 final class IteratorFilter extends RecursiveFilterIterator implements RecursiveIterator
 {
```

Reply:
Thank you, that works.